satpy/readers/hrit_base.py

```diff
@@ -320,11 +321,12 @@ def read_band(self, key, info):
         elif self.mda['number_of_bits_per_pixel'] in [8, 10]:
             dtype = np.uint8
         shape = (shape, )
-        data = np.memmap(self.filename, mode='r',
-                         offset=self.mda['total_header_length'],
-                         dtype=dtype,
-                         shape=shape)
-        data = da.from_array(data, chunks=shape[0])
+        with utils.unzip_context(self.filename) as fn:
+            data = np.memmap(fn, mode='r',
+                             offset=self.mda['total_header_length'],
+                             dtype=dtype,
+                             shape=shape)
+            data = da.from_array(data, chunks=shape[0])
```

Comment:
I know we talked about this before, but I'm curious how much difference it makes in performance to unzip the file and do a memmap, vs using the bz2 open and frombuffer. Did you check it out?

Reply:
Great, thanks a lot!

Reply:
According to the comparisons, I think we should keep it as you wrote.